src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.OpenSsl.cs

```diff
@@ -385,25 +389,31 @@ internal static SecurityStatusPalErrorCode DoSslHandshake(SafeSslHandle context,
                 }
             }
 
-            int retVal = Ssl.SslDoHandshake(context);
-            if (retVal != 1)
+            try
             {
-                Exception? innerError;
-                Ssl.SslErrorCode error = GetSslError(context, retVal, out innerError);
-
-                if (error == Ssl.SslErrorCode.SSL_ERROR_WANT_X509_LOOKUP)
+                int retVal = Ssl.SslDoHandshake(context);
+                if (retVal != 1)
                 {
-                    return SecurityStatusPalErrorCode.CredentialsNeeded;
-                }
+                    Exception? innerError = null;
+                    Ssl.SslErrorCode error = GetSslError(context, retVal, out innerError);
```

Comment:
Shouldn't need to initialize this to `null`. Or, you can just fold the two lines:
```suggestion
                    Ssl.SslErrorCode error = GetSslError(context, retVal, out Exception? innerError);
```